src/bundle/Resources/public/js/CKEditor/core/base-ckeditor.js

```diff
@@ -0,0 +1,111 @@
+(function(global, doc, eZ, CKEditor) {
+    class BaseRichText {
+        constructor(config) {
+            this.ezNamespace = 'http://ez.no/namespaces/ezpublish5/xhtml5/edit';
+            this.xhtmlNamespace = 'http://www.w3.org/1999/xhtml';
+
+            this.editor = null;
+
+            this.xhtmlify = this.xhtmlify.bind(this);
+            this.getData = this.getData.bind(this);
+        }
+
+        getData() {
+            return this.editor.getData();
+        }
+
+        getHTMLDocumentFragment(data) {
+            const fragment = doc.createDocumentFragment();
+            const div = fragment.ownerDocument.createElement('div');
+            const parsedHTML = new DOMParser().parseFromString(data, 'text/xml');
+            const importChildNodes = (parent, element, skipElement) => {
+                let i;
+                let newElement;
+
+                if (skipElement) {
+                    newElement = parent;
+                } else {
+                    if (element.nodeType === Node.ELEMENT_NODE) {
+                        newElement = parent.ownerDocument.createElement(element.localName);
+
+                        for (i = 0; i !== element.attributes.length; i++) {
```

Comment:
why not `for (let i = 0; i !== element.attributes.length; i++) {`? Why declaring this variable at the beginnign?

Reply:
I think someone suggested this because this `i` is used in two loops.